internal/services/jsonrpc/usersvc/service.go

```diff
@@ -519,6 +592,13 @@ func (svc *Service) Call(ctx context.Context, req *userjson.CallRequest) (*userj
 
 	}
 
+	// If authenticateRPCs is enabled, we need to check the challenge
+	if svc.authenticateRPCs {
+		if err := svc.verifyChallenge(msg.Challenge); err != nil {
```

Comment:
It seems like the most important detail to verify here (besides validity of signature) is that the sender in the challenge is the sender in the call message.  But we're missing that

In context of https://github.com/kwilteam/kwil-db/pull/946/files#r1739103177, this is the only concern of this Call method, if it were the job of the server to verify the signature and the pass the verified identity to the service method to do what it will: (1) in call, ensure that the call.sender is equal to challenge.sender, and (2) in query, just make sure that there is an authenticated sender.

BTW, what's the point of authentication on `query` if it doesn't matter who signed, only it was signed?  Don't we want a whitelist or something?  Or it it supposed to only allow the schema owner to do ad hoc queries?

Reply:
talked to brennan about this. Apparently, the expectation is, we shouldn't allow adhoc queries in private mode.

Reply:
Ahhh, that makes more sense.  As it was, verifying that there was a valid signature with no requirement that signer match anything in particular... well that didnt' do much. :)